### ISSUE TITLE
Code completion does not work on pandas dataframes loaded from CSV

### ISSUE THREAD

**Issue by Hamulus** [spyder-ide/spyder#5998]
## Description of your problem

**What steps will reproduce the problem?**

1. create an empty or some other dataframe to make sure that code completion works at all
`df = pd.DataFrame()`
so when writing
`df.`
the code completion works
this step is not required, I made it only to make sure that code completion works on pandas dataframes
2. load CSV to this dataframe
`df = pd.read_csv('mysample.csv')`
or to a new dataframe
`df2 = pd.read_csv('mysample.csv')`
3. and when writing the name of dataframe with dot
`df.`
or
`df2.`
code completion does not open automatically and cannot be opened by Tab or Ctrl+Space

Also function docs do not appear on dataframes loaded from CSV

So I found the only one way to make code completion work:
`df = pd.DataFrame(df)`
after CSV loading

**What is the expected output? What do you see instead?**
I expect code completion dropdown but I do not see it

## Versions and main components

* Spyder Version: 3.2.4
* Python Version: 3.6.3
* Qt Version:5.6.2
* PyQt Version:5.6
* Operating system: Windows 8.1

## Dependencies

IPython >=4.0    :  6.1.0 (OK)
cython >=0.21    :  0.26.1 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
numpy >=1.7      :  1.13.3 (OK)
pandas >=0.13.1  :  0.21.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.6.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.7.4 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
rope >=0.9.4     :  0.10.5 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
sympy >=0.7.3    :  1.1.1 (OK)

**Comment by CAM-Gerlach** [comment]
Tested to confirm and expand upon the behavior. With

```
df = pd.read_csv(PATH_TO_CSV)
```

I experienced the above—dot and tab completion and help retrieval do not work in the Editor. However, tab completion and help retrieval _do_ work in the Console (dot completion does not work in the Console for anything, as far as I'm aware). For the following control cases, everything works as expected:

```
df0 = pd.DataFrame()
df1 = pd.DataFrame({'col1': [1, 2, 3], 'col2': ["a", "b", "c"]})
df2 = pd.DataFrame(np.random.randn(10, 5))
df3 = pd.DataFrame(np.random.randn(10, 5),
                columns=['a', 'b', 'c', 'd', 'e'])
```

Therefore, the problem depends on how code introspection (e.g. `jedi`, `rope`) work differently in the Editor vs. the Console; if I understand correctly, the former effectively does static analysis of the code while the latter looks at what's actually loaded in the environment. Not 100% sure something like this is 100% within Spyder's control vs. being a limitation with our completion libraries or `pandas`, but there's certainly a chance.

@ccordoba12 ?
